compiler/rustc_const_eval/src/interpret/terminator.rs

```diff
@@ -503,6 +503,25 @@ impl<'mir, 'tcx: 'mir, M: Machine<'mir, 'tcx>> InterpCx<'mir, 'tcx, M> {
                     }
                 }
 
+                let attrs = self.tcx.codegen_fn_attrs(instance.def_id());
```

Comment:
Please add a comment explaining what happens here.  If the number of variables shared with the context is low, can you factor this into a separate function? This function is already super long, so it'd be better if we can prevent it from growing further. :)

Reply:
It only needs `self` and `instance`, so consider it done.